### ISSUE TITLE
Supporting timecode of 24:00:00:00 upwards in MOV/MP4 timecode tracks

### ISSUE THREAD

**Issue by vectronic** [MediaArea/MediaInfoLib#1248]
Hello,

I've come across an interesting MOV file which has a timecode track indicating start time of `24:00:00:00`.

In ffprobe, the start timecode is output as `24:00:00:00`, however in MediaInfoLib it is reported as `00:00:00:00`.

I have narrowed the difference down to the following change I have tried:

https://github.com/vectronic/MediaInfoLib/commit/8dd12dd1a5d0921fa5bc12fc47198d82a107234d

I realise this change is not usable in it's current form as it doesn't account for numbers greater than 99 taking up more than 2 digits in timecode format.

I have looked into the specs for mov/mp4 and can't see anything which says the timecode can't be `24:00:00:00` or greater and I can see `24HourMax=No`.

With the above change the timecode tracks outputs as:

```
Time code of first frame                 : 24:00:00:00
```

One thing I have not been able to solve is that even with the above change, the start time code for the video track is still output as: 

```
Time code of first frame                 : 00:00:00:00
```

I believe this should refer to the timecode track?

It looks like at some point this value is ignored despite the following appearing in the MediaInfoLib full output for the video track:

```
Delay                                    : 86400000
Delay                                    : 24 h 0 min
Delay                                    : 24 h 0 min 0 s 0 ms
Delay                                    : 24 h 0 min
Delay                                    : 24:00:00.000
```

and the following in the traces (where the DTS is calculated based on the timecode):

```
00004672   slice_start - Frame (decoding order) 0 - Frame (presentation order) 2 - picture_coding_type I - temporal_reference 2 - DTS 24:00:00.000 - time_code 0:0:0.80 (2178 bytes)
```

In summary I guess I am asking:

* Is the limiting to timecode below 24h intentional? If not, would you be happy for a PR with better handling of hours above 99?
* Am I on the wrong path thinking that the video start time code should also display `24:00:00.000` ? If I am not wrong, do you have any pointers to where a fix could be made for this?

I have attached full output of the file before and after the change as well as a full parsing trace output (without the change).

[example-output.txt](https://github.com/MediaArea/MediaInfoLib/files/4523695/example-output.txt)

[example-output-after-change.txt](https://github.com/MediaArea/MediaInfoLib/files/4523694/example-output-after-change.txt)

[mediainfo-trace.txt](https://github.com/MediaArea/MediaInfoLib/files/4523585/mediainfo-trace.txt)

**Comment by JeromeMartinez** [comment]
> Is the limiting to timecode below 24h intentional? 

It was a lazy method for supporting time codes we were using ourselves, no other intention.

> If not, would you be happy for a PR with better handling of hours above 99?

Yes.
Better to show the real content of the file, even if there is a mismatch between time codes after that (see below).

> Am I on the wrong path thinking that the video start time code should also display 24:00:00.000 ?

This time code comes from MPEG Video stream, from your trace:
```
00004544   group_start - 0:0:0.0 (8 bytes)
00004544    Header (4 bytes)
00004544     synchro:                            1 (0x000001)
00004547     start_code:                         184 (0xB8)
00004548    time_code_drop_frame_flag:           No
00004548    time_code_time_code_hours:           0 (0x00) - (5 bits)
00004548    time_code_time_code_minutes:         0 (0x00) - (6 bits)
00004549    time_code_time_code_seconds:         0 (0x00) - (6 bits)
0000454A    time_code_time_code_pictures:        0 (0x00) - (6 bits)
0000454B    closed_gop:                          Yes
0000454B    broken_link:                         No
```

So we show this value, and this value is not supporting \>= 24:00:00:00 time codes... So it should stay as is, because  different time codes are the content of your file with your modification.
\>= 24:00:00:00 time codes are not so good ;-)

The delay thing is there for very cray reasons (me hiding...) for a specific big sponsors, syncing with the container time code, so the behavior is intentional (I should clean up that at some point...)

**Comment by vectronic** [comment]
Ok - that all makes sense - and thanks for the quick response.

If I was to make a change to for timecode calculation, I must admit I don't know the best approach :-)

```
if hours > 99
   hours = hours % 24
```

?!

**Comment by JeromeMartinez** [comment]
It is a lazy method :-D for handling 24:00:00:00 instead of adding digits to the hour counts in the string output, but it is fair enough for handling the issue you have.

Just an issue with storing the intermediate value in `hours` directly, I disagree as "256" will become 0 with your patch (`hours` is a 8-bit unsigned), please use an intermediate 32-bit value, something like

```
int64s Temp = (((Frames_ / FramesPerSecond_) / 60) / 60);
Hours = (Temp>99 || Temp<-99)?(Temp%24):Temp;
```
(I just remind that negative time codes sometimes happen)
